### ISSUE TITLE
Provide more complete installation instructions

### ISSUE THREAD

**Issue by kopeboy** [cosmos/gaia#2775]
## Summary

Two problems:

1. If you have a newer go version, `make install` will fail with `ERROR: Go version 1.20 is required for v13.0.1 of Gaia.`
I tried downloading & installing go1.20 (as an addition to my go1.21.3) but then I don't know how to tell gaia to use that.

2. If you try by downloading the binary for macOS (darwin arm64) then the os doesn't know how to open it and no instructions are given. May it be a missing file extension? I don't know, I'm a noob. 

## Problem Definition

Non-expert developers and just users might want to use the gaia daemon CLI to interact with the chain, since no GUI wallet provides interfaces for all the features, hence more ELI5 are needed imho.

## Proposal

Improve intallation documentation for common platforms like macOS and Windows by providing solution instructions to common errors like those I just encountered.

Thank you
____

#### For Admin Use

- [ ] Not duplicate issue
- [ ] Appropriate labels applied
- [ ] Appropriate contributors tagged
- [ ] Contributor assigned/self-assigned
- [ ] Is a spike necessary to map out how the issue should be approached?

**Comment by kopeboy** [comment]
(Note for other noobs: you can download the Source code archive, zip or tar.gz, and just open that)

**Comment by mmulji-ic** [comment]
Hi @kopeboy thanks for reporting this issue.

Regarding item 1:
I've updated the release details - which were missing this information, usually the go version should be included as well, as its quite important for the good functioning of the network - thats why we mandate it here in the makefile. 

Other onboarding/getting started docs should also have this information. Go usually works on a system level rather than via simple configuration. So you need to install the correct version of go or use docker with the go version that you want to use to compile the source code if you do not want to make system level changes just for one application.

Regarding item 2:
Typically, MacOS/Linux binaries do not have extensions. If your making from source you should be able to run the application directly.

We can expand installation instructions as part of this issue as well.
